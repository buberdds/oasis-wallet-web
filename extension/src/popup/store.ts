import { Store, applyMiddleware } from 'webext-redux'
import createSagaMiddleware from 'redux-saga'

export default function configureAppStore() {
  const store = new Store()
  const sagaMiddleware = createSagaMiddleware({})
  const middlewares = [sagaMiddleware]
  const storeWithMiddleware = applyMiddleware(store, ...middlewares)
  return storeWithMiddleware
}
