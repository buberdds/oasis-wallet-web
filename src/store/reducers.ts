/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers } from '@reduxjs/toolkit'
import createWalletReducer from 'app/pages/CreateWalletPage/slice'
import openWalletReducer from 'app/pages/OpenWalletPage/slice'
import accountReducer from 'app/state/account'
import fatalErrorReducer from 'app/state/fatalerror'
import ledgerReducer from 'app/state/ledger'
import networkReducer from 'app/state/network'
import stakingReducer from 'app/state/staking'
import transactionReducer from 'app/state/transaction'
import walletReducer from 'app/state/wallet'
import themeReducer from 'styles/theme/slice'

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export function createReducer(injectedReducers = {}) {
  // Initially we don't have any injectedReducers, so returning identity function to avoid the error
  const rootReducer = combineReducers({
    ...injectedReducers,
    account: accountReducer,
    createWallet: createWalletReducer,
    fatalError: fatalErrorReducer,
    ledger: ledgerReducer,
    network: networkReducer,
    openWallet: openWalletReducer,
    staking: stakingReducer,
    theme: themeReducer,
    transaction: transactionReducer,
    wallet: walletReducer,
  })

  return rootReducer
}
