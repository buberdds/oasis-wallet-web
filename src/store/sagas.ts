import { all } from 'typed-redux-saga'

import { accountSaga } from 'app/state/account/saga'
import { networkSaga } from 'app/state/network/saga'
import { ledgerSaga } from 'app/state/ledger/saga'
import { stakingSaga } from 'app/state/staking/saga'
import { transactionSaga } from 'app/state/transaction/saga'
import { rootWalletSaga } from 'app/state/wallet/saga'

export default function* rootSagas() {
  yield all([accountSaga(), networkSaga(), ledgerSaga(), stakingSaga(), transactionSaga(), rootWalletSaga()])
}
