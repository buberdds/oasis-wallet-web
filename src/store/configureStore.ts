import { configureStore } from '@reduxjs/toolkit'
import createSagaMiddleware from 'redux-saga'

import { createReducer } from './reducers'
import rootSagas from './sagas'
import { RootState } from 'types'
import { fatalErrorActions } from 'app/state/fatalerror'

export function configureAppStore(state?: Partial<RootState>) {
  const sagaMiddleware = createSagaMiddleware({
    onError: (error, info) => {
      store.dispatch(
        fatalErrorActions.setError({
          message: error.toString(),
          stack: error.stack,
          sagaStack: info.sagaStack,
        }),
      )
    },
  })

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware]

  const store = configureStore({
    reducer: createReducer(),
    middleware: getDefaultMiddleware => getDefaultMiddleware().concat(middlewares),
    devTools:
      /* istanbul ignore next line */
      process.env.NODE_ENV !== 'production',
    preloadedState: state,
  })

  sagaMiddleware.run(rootSagas)

  return store
}
