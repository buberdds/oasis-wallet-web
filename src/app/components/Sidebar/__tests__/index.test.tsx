import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import * as React from 'react'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'

import { Navigation } from '..'

const renderComponent = (store: any) => {
  return render(
    <Provider store={store}>
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>
    </Provider>,
  )
}

describe('<Navigation />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
    jest.resetAllMocks()
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it.todo('should be responsive')
})
