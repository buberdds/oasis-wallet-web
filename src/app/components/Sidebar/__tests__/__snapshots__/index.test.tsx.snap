// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should match snapshot 1`] = `
.c10 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c10 g {
  fill: inherit;
  stroke: inherit;
}

.c10 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c10 *[stroke*="#"],
.c10 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c10 *[fill-rule],
.c10 *[FILL-RULE],
.c10 *[fill*="#"],
.c10 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background: component-sidebar;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100%;
  width: 220px;
  padding-top: 24px;
  padding-bottom: 24px;
  padding-left: 0px;
  padding-right: 0px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-bottom: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: url(IMAGE_MOCK) no-repeat center center;
  background-size: cover;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 48px;
  width: 48px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 100%;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  padding: 0px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background: background-oasis-blue;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 24px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-top: 12px;
  padding-bottom: 12px;
  padding-left: 24px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  padding: 12px;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c6 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 12px;
}

.c11 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 24px;
}

.c5 {
  font-size: 18px;
  line-height: 24px;
}

.c13 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  width: 100%;
}

.c13:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c13:focus > circle,
.c13:focus > ellipse,
.c13:focus > line,
.c13:focus > path,
.c13:focus > polygon,
.c13:focus > polyline,
.c13:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c13:focus::-moz-focus-inner {
  border: 0;
}

.c13:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c13:focus:not(:focus-visible) > circle,
.c13:focus:not(:focus-visible) > ellipse,
.c13:focus:not(:focus-visible) > line,
.c13:focus:not(:focus-visible) > path,
.c13:focus:not(:focus-visible) > polygon,
.c13:focus:not(:focus-visible) > polyline,
.c13:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c13:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c16 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c16:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus > circle,
.c16:focus > ellipse,
.c16:focus > line,
.c16:focus > path,
.c16:focus > polygon,
.c16:focus > polyline,
.c16:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c16:focus::-moz-focus-inner {
  border: 0;
}

.c16:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible) > circle,
.c16:focus:not(:focus-visible) > ellipse,
.c16:focus:not(:focus-visible) > line,
.c16:focus:not(:focus-visible) > path,
.c16:focus:not(:focus-visible) > polygon,
.c16:focus:not(:focus-visible) > polyline,
.c16:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c16:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-top: 12px;
    padding-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-left: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-right: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c8 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c18 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c19 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c4 {
    width: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    height: 6px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      href="/"
    >
      <div
        class="c2"
      >
        <div
          class="c3 StyledAvatar-sc-1suyamb-1"
        />
        <div
          class="c4"
        />
        <span
          class="c5"
        >
          Oasis Wallet
        </span>
      </div>
    </a>
  </div>
  <div
    class="c6"
  />
  <div
    class="c7"
  >
    <nav
      class="c8"
    >
      <a
        aria-current="page"
        aria-label="menu.home"
        class="active"
        data-testid="nav-home"
        href="/"
      >
        <div
          class="c9"
        >
          <svg
            aria-label="Home"
            class="c10"
            viewBox="0 0 24 24"
          >
            <path
              d="m1 11 11-9 11 9m-8 12v-8H9v8m-5 0V9m16 14V9"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
          <div
            class="c11"
          />
          <span
            class="c5"
          >
            menu.home
          </span>
        </div>
      </a>
      <div
        class="c6"
      />
      <div
        class="c6"
      />
    </nav>
  </div>
  <div
    class="c6"
  />
  <nav
    class="c12"
  >
    <button
      aria-label="theme.darkMode"
      class="c13"
      type="button"
    >
      <div
        class="c14"
      >
        <svg
          aria-label="Moon"
          class="c10"
          viewBox="0 0 24 24"
        >
          <path
            d="M9.874 5.008c2.728-1.68 6.604-1.014 8.25.197-2.955.84-5.11 3.267-5.242 6.415-.18 4.28 3.006 6.588 5.24 7.152-1.964 1.343-4.36 1.293-5.235 1.172-3.568-.492-6.902-3.433-7.007-7.711-.106-4.278 2.573-6.35 3.994-7.225z"
            stroke="#000"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <div
          class="c11"
        />
        <span
          class="c5"
        >
          theme.darkMode
        </span>
      </div>
    </button>
    <div
      class="c6"
    />
    <div
      class="c6"
    />
    <div
      class="c15"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="Open Menu"
        class="c16"
        type="button"
      >
        <div
          class="c17"
        >
          <div
            class="c18"
          >
            <svg
              aria-label="Language"
              class="c10"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 23c6.075 0 11-4.925 11-11S18.075 1 12 1 1 5.925 1 12s4.925 11 11 11zm0 0c3 0 4-5 4-11S15 1 12 1 8 6 8 12s1 11 4 11zM2 16h20M2 8h20"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="c19"
          >
            <span
              class="c5"
            >
              Language
            </span>
          </div>
          <div
            class="c18"
          >
            <svg
              aria-label="FormDown"
              class="c10"
              viewBox="0 0 24 24"
            >
              <path
                d="m18 9-6 6-6-6"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </button>
    </div>
    <div
      class="c6"
    />
    <a
      aria-label="GitHub"
      href="https://github.com/oasisprotocol/oasis-wallet-web"
      rel="noopener"
      target="_blank"
    >
      <div
        class="c14"
      >
        <svg
          aria-label="Github"
          class="c10"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.999 1C5.926 1 1 5.925 1 12c0 4.86 3.152 8.983 7.523 10.437.55.102.75-.238.75-.53 0-.26-.009-.952-.014-1.87-3.06.664-3.706-1.475-3.706-1.475-.5-1.27-1.221-1.61-1.221-1.61-.999-.681.075-.668.075-.668 1.105.078 1.685 1.134 1.685 1.134.981 1.68 2.575 1.195 3.202.914.1-.71.384-1.195.698-1.47-2.442-.278-5.01-1.222-5.01-5.437 0-1.2.428-2.183 1.132-2.952-.114-.278-.491-1.397.108-2.91 0 0 .923-.297 3.025 1.127A10.536 10.536 0 0 1 12 6.32a10.49 10.49 0 0 1 2.754.37c2.1-1.424 3.022-1.128 3.022-1.128.6 1.514.223 2.633.11 2.911.705.769 1.13 1.751 1.13 2.952 0 4.226-2.572 5.156-5.022 5.428.395.34.747 1.01.747 2.037 0 1.47-.014 2.657-.014 3.017 0 .295.199.637.756.53C19.851 20.979 23 16.859 23 12c0-6.075-4.926-11-11.001-11"
            fill="#333"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="c11"
        />
        <span
          class="c5"
        >
          GitHub
        </span>
      </div>
    </a>
  </nav>
</div>
`;
